recipes/newrelic/apm/java/linux.yml

```diff
@@ -49,18 +58,31 @@ preInstall:
       apt-get install -y -qq ./nri-introspector-java_0.1.0_SNAPSHOT-2_amd64.deb 2> /dev/null || true
     fi
 
-    nri-lsi-java -list | tr -d '[]' | tr ',' '\n' > $tmpDir/processes
-    for JAVA_PID in $(cat $tmpDir/processes)
+    # check if any tomcat instances are service-managed
+    nri-lsi-java -list | tr -d '[]' | tr ',' '\n' > $TMP_DIR/all_java_processes
+    for JAVA_PID in $(cat $TMP_DIR/all_java_processes)
     do
       introspectionData=$(nri-lsi-java -introspect ${JAVA_PID})
-      mainClass=$( echo "$introspectionData" | grep -oP '.*\K(?<=mainClass":").*?(?=\")' )
-      if [[ "$mainClass" == "org.apache.catalina.startup.Bootstrap" ]]; then
-        rm -rf $tmpDir
-        exit 0
+      MAIN_CLASS=$( echo "$introspectionData" | grep -oP '.*\K(?<=mainClass":").*?(?=\")' )
```

Comment:
@mbruzina this is a gnarly regex, which I don't doubt does what we need to do. Wouldn't this work the same if we were just grepping on "catalina" or even "org.apache.catalina.startup.Bootstrap" if we wanted to be more accurate? I'm not sure we care much about parsing what's returned than making sure there is a catalina somewhere in the content?